types/openpgp/index.d.ts

```diff
@@ -4899,7 +4912,15 @@ export interface EncryptResult {
  *          sessionKey: { data, algorithm, aeadAlgorithm } (if `returnSessionKey` was true)
  *          }
  */
-export function encrypt(options: EncryptOptions): Promise<EncryptResult>;
+export function encrypt(options: EncryptOptions & { armor?: true, detached?: false }): Promise<EncryptResult & { data: string }>
+export function encrypt(options: EncryptOptions & { armor?: true, detached: true }): Promise<EncryptResult & { data: string, signature: string }>
+export function encrypt(options: EncryptOptions & { armor: false, detached?: false }): Promise<EncryptResult & { message: message.Message }>
+export function encrypt(options: EncryptOptions & { armor: false, detached: true }): Promise<EncryptResult & { message: message.Message, signature: signature.Signature }>
+export function encrypt(options: EncryptOptions): Promise<EncryptResult & {
+    data: string | ReadableStream<String>;
+    message: message.Message;
+    signature: string | ReadableStream<String> | signature.Signature;
+}>;
```

Comment:
Why not just leave data, message, and signature as optional properties of EncryptResult?